Relax licensing terms and add explicit license for header files

kholia opened the pull request:
Hi,

We are adding support for CH32 RISC-V MCUs in Zephyr via https://github.com/zephyrproject-rtos/zephyr/pull/73761.

The following licensing text in this repository is NOT compatible with Zephyr.

```
Attention: This software (modified or not) and binary are used for 
microcontroller manufactured by Nanjing Qinheng Microelectronics.
```

This PR removes this restrictive licensing condition and explicitly licenses the header files in this repository under a dual-license (MIT License and Apache-2.0 License). Please note that many files in this repository are already licensed under Apache-2.0 License.

CC  @openwch @fabiobaltieri @nzmichaelh @AlexanderMandera @cnlohr.

AlexanderMandera commented:
Here is the quoted message from Patrick Yang to cnlohr about the licensing, as a reference:
https://github.com/zephyrproject-rtos/zephyr/pull/73761#issuecomment-2188506016

cnlohr commented:
This re-license affects source files, not just header files. All that is important is that the header files are permitted for use in other licensed projects.

kholia commented:
@cnlohr Thanks! I have updated this PR to only target the header files now.

Furthermore, I have added the following file to this repository.

```
$ cat LICENSE 
The header files in this repository are dual-licensed under MIT License and
Apache-2.0 License.

Copyright (c) 2021 Nanjing Qinheng Microelectronics Co., Ltd.
```

zhang-wenchao commented:
Any update? Please merge this PR. Thank you!

cnlohr commented:
@zhang-wenchao merging here is no longer required.  They gave explicit license permission for the header files externally.
